### PR TITLE
Error handling revamp

### PR DESCRIPTION
**Pull request by Aberdeener**
Screen Shot 2021-03-06 at 4 59 12 PM

-Clickable frames for each file and line that was called before the error

Currently doesn't catch all errors and if an error happens before the class is loaded bad things happen. I think it'll need a revamp to catch everything.

Still lots to do in terms of QOL, design and general impl.
Would appreciate reviews, dont wanna keep working if there is anything fatal anyone else disagrees with

### CODE REVIEW

**Comment by Aberdeener** [comment]
Updated style: (obviously scroll down to see the highlighted line of code + click on other frames)
Screen Shot 2021-03-08 at 10 35 30 PM

**Comment by Aberdeener** [comment]
Marking as ready for review, I can finish the details this evening before merging.

File: core/init.php [hunk 1]
```diff
@@ -11,51 +11,55 @@
 
 // Nameless error handling
 require_once(join(DIRECTORY_SEPARATOR, array(ROOT_PATH, 'core', 'classes', 'ErrorHandler.php')));
-set_error_handler("ErrorHandler::catchError");
-register_shutdown_function("ErrorHandler::catchFatalError");
+set_exception_handler([ErrorHandler::class, 'catchException']);
```
Comment (samerton) [inline core/init.php:14]: Does this replace the need for `register_shutdown_function`?

File: core/init.php [hunk 2]
```diff
@@ -11,51 +11,55 @@
...
+// Any errors thrown now will not be caught by the error handler, as they require some classes be preloaded
 spl_autoload_register(function ($class) {
     $path = join(DIRECTORY_SEPARATOR, array(ROOT_PATH, 'core', 'classes', $class . '.php'));
-    if (file_exists($path)) require_once($path);
+    if (file_exists($path)) {
+        require_once($path);
+    }
 });
-
+
```
Comment (samerton) [inline core/init.php:62]: Extra spacing 😄

File: core/init.php [hunk 3]
```diff
@@ -65,8 +69,8 @@
 
 if ($page != 'install') {
     /*
-     *  Initialise
-     */
+    *  Initialise
```
Comment (samerton) [inline core/init.php:72]: Can we keep the asterisks inline? They look neater imo

File: core/init.php [hunk 4]
```diff
@@ -157,9 +163,9 @@
             ini_set('display_errors', 0);
         }
     }
-	
-	// Configurations
-	$configuration = new Configuration($cache);
+
```
Comment (samerton) [inline core/init.php:166]: Extra space

File: modules/Core/init.php [hunk 5]
```diff
@@ -19,4 +19,7 @@
 
 require_once(ROOT_PATH . '/modules/Core/module.php');
 
+// TODO: start_line is negative when issue LOC is less than line_buffer
```
Comment (samerton) [inline modules/Core/init.php:22]: Is this still needed?

**Review by samerton — CHANGES_REQUESTED** [review]

File: core/init.php (hunk 3 above)
Comment (Aberdeener) [reply core/init.php:72]: Good catch, must have been a weird copy/paste issue 😄

File: core/assets/css/prism.css [hunk 6]
```diff
@@ -0,0 +1,278 @@
+/* PrismJS 1.23.0
```
[the remark this answers is not in the document]
Comment (Aberdeener) [reply core/assets/css/prism.css]: Thanks, fixed

File: core/init.php (hunk 1 above)
Comment (Aberdeener) [reply core/init.php:14]: Good point, I misunderstood the documentation. 
I have added it back with a new function.

**Comment by Aberdeener** [comment]
Good to go for me now. See #development channel for some notes on latest commit. Thanks!

**Pull request merged**